### PR TITLE
Fix 'anyShipping' logic for gather-drop

### PR DESCRIPTION
**Pull request by hoyosjs**
Looks like at some point the `release.json` stopped populating the products node with the paths of the shipping assets. Turns out it only adds products it thinks has shipping assets, which wasn't being bubbled up. I added the extra download assets to be considered as well, let me know if that is not right.

### CODE REVIEW

**Comment by hoyosjs** [comment]
@MattGal @mmitche @adiaaida, as I can't add reviewers here.

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs
Comment (adiaaida) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs:840]: I don't think we care about anyShipping for the extra assets, which is probably how this bool went missing in the first place. I think this can be removed.

**Review by adiaaida — APPROVED** [review]
One little note, otherwise, LGTM. Thanks for this fix!

**Review by MattGal — APPROVED** [review]
I defer to @adiaaida  on this subject so if she signs off it's probably all good.

**Comment by hoyosjs** [comment]
Feel free to merge whenever appropriate. I don't have write access.

**Pull request merged**
